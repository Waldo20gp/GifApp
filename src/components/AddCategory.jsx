import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState("")

    const onInputChange = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmit= ( event ) => {
        event.preventDefault();
        const inputValueTrim = inputValue.trim(); 
        if(inputValueTrim.length<1) return;
        onNewCategory(inputValueTrim)
        setInputValue("")
    }

    return (
        <form onSubmit={ onSubmit }>
            <input 
                type="text"
                placeholder="Buscar gif" 
                value={inputValue} 
                onChange={onInputChange} 
            />
        </form>
    )
}


AddCategory.propTypes = {
    onNewCategory:PropTypes.func.isRequired
}