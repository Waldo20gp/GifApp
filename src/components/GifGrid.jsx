import { GifGridItem } from "./GifGridItem";
import { useState, useEffect } from "react";

import { useFetchGifs } from "../hooks/useFetchGifs";
import { getGifs } from "../helpers/getGifs";

export const GifGrid = ({ category }) => {


    //  const { images, isLoading } = useFetchGifs();
  const [images, setImages] = useState([])

  const getImages = async(category) => {
    const images = await getGifs(category);
    setImages(images);
  }


  useEffect(() => {
    getImages(category)
  }, [])

  

  return (
    <>
        <h3>{ category }</h3>
        <div className="card-grid">
          {images.map((image)=>(
            <GifGridItem 
            key={image.id}
            { ...image }
            />
            ))}
        </div>
    </>
  )
}



