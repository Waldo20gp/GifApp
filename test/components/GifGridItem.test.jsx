import { cleanup, render, screen } from "@testing-library/react"
import { GifGridItem } from "../../src/components/GifGridItem"

describe('Pruebas en el componente GifGridItem', () => { 
    afterEach(cleanup)
    
    const url="http://urltest.com/";
    const title = "titleTest"

    test('Debe asegurarse el snapshot es correcto', () => { 
       const {container} = render(<GifGridItem url={url} title={title} />)

       expect(container).toMatchSnapshot();
    });

    test('Debe de mostrat la imagen con el URL y el ALT indicado', () => { 
        render(<GifGridItem url={url} title={title} />);
        screen.debug()

        const { src, alt } = screen.getByRole('img');

        expect( src ).toBe(url);
        expect( alt ).toBe(title);

        // expect(screen.getByRole('img').src).toBe(url.toLowerCase());
     });

     test('Se debe mostrar el titulo en el componente', () => { 
        render(<GifGridItem url={url} title={title} />);
        expect(screen.findByText(title)).toBeTruthy()
      })



 })