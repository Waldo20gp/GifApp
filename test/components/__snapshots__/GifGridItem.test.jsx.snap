// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGridItem Debe asegurarse el snapshot es correcto 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titleTest"
      src="http://urltest.com/"
    />
    <p>
      titleTest
    </p>
  </div>
</div>
`;
